.evg.yml

```diff
@@ -662,14 +667,19 @@ functions:
         file: mongo-jdbc-driver/artifacts/expansions.yml
 
   "fetch jdbc files":
+    - command: ec2.assume_role
+      params:
+        role_arn: ${assume_role_arn}
+        duration_seconds: 3600
     - command: s3.get
       params:
         working_dir: mongo-jdbc-driver
-        aws_key: ${aws_key}
-        aws_secret: ${aws_secret}
-        local_file: mongo-jdbc-driver/build/libs/mongodb-jdbc-${MDBJDBC_VER}.jar
-        remote_file: ${S3_ARTIFACTS_DIR}/mongodb-jdbc-${MDBJDBC_VER}.jar
-        bucket: mciuploads
+        aws_key: ${AWS_ACCESS_KEY_ID}
+        aws_secret: ${AWS_SECRET_ACCESS_KEY}
+        aws_session_token: ${AWS_SESSION_TOKEN}
+        local_file: mongo-jdbc-driver/build/libs/mongodb-jdbc-eap-${MDBJDBC_VER}.jar
+        remote_file: ${S3_ARTIFACTS_DIR}/mongodb-jdbc-eap-${MDBJDBC_VER}.jar
+        bucket: mongo-jdbc-driver-eap
```

Comment:
We repeat this same structure many times now, so let's use a YAML anchor and aliases to reduce it. I think something like this should work:
```
# towards the top of the file
variables:
  - &assume_role_cmd
    command: ec2.assume_role
    params:
      role_arn: ${assume_role_arn}
      duration_seconds: 3600
  - &aws_std_config
    aws_key: ${AWS_ACCESS_KEY_ID}
    aws_secret: ${AWS_SECRET_ACCESS_KEY}
    aws_session_token: ${AWS_SESSION_TOKEN}
    bucket: evg-bucket-mongo-jdbc-driver

...

# In each of these functions...
  "fetch jdbc files":
    - *assume_role_cmd
    - command: s3.get
      params:
        <<: *aws_std_config
        working_dir: mongo-jdbc-driver
        local_file: ...
        remote_file: ...

  "upload unit test classes":
    - *assume_role_cmd
    - command: s3.put
      params:
        <<: *aws_std_config
        ...
        
...
```

Admittedly, I did not test exactly this code but the rough idea should work, potentially with some minor tweaks. This should really help us not only cut down on the size of this config file but also avoid issues in the future where we have to make changes to each of these affected functions in bulk.

Reply:
Good idea, I'll experiment and try to get this working. Also glad you saw the bucket update, this looks like the missing piece we were talking about yesterday :)

Reply:
Based on this part of the evergreen docs: https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-Configuration-Files#functions ! Could be a useful reference 👍

Reply:
definitely, thanks for linking. I'll do the same in the parallel change to odbc-eap